plugins/inputs/mikrotik/mikrotik.go

```diff
@@ -0,0 +1,224 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mikrotik
+
+import (
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	common_tls "github.com/influxdata/telegraf/plugins/common/tls"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Mikrotik struct {
+	Address         string          `toml:"address"`
+	IgnoreCert      bool            `toml:"ignore_cert,omitempty"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	IgnoreComments  []string        `toml:"ignore_comments"`
+	IncludeModules  []string        `toml:"include_modules"`
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Log             telegraf.Logger `toml:"-"`
+	common_tls.ClientConfig
+
+	tags map[string]string
+
+	url []mikrotikEndpoint
+
+	client *http.Client
+
+	systemTagsURL []string
+}
+
+func (*Mikrotik) SampleConfig() string {
+	return sampleConfig
+}
+
+func (h *Mikrotik) Start() error {
+	return h.getSystemTags()
+}
+
+func (h *Mikrotik) Init() error {
+	if h.Username.Empty() {
+		return errors.New("mikrotik init -> username must be present")
+	}
+
+	if len(h.IncludeModules) == 0 {
+		h.IncludeModules = append(h.IncludeModules, "system_resourses")
+	}
+
+	mainPropList, systemResourcesPropList, systemRouterBoardPropList := createPropLists()
+
+	h.systemTagsURL = []string{
+		h.Address + "/rest/system/resource?" + systemResourcesPropList,
+		h.Address + "/rest/system/routerboard?" + systemRouterBoardPropList,
+	}
+
+	for _, selectedModule := range h.IncludeModules {
+		if _, ok := modules[selectedModule]; !ok {
+			return fmt.Errorf("mikrotik init -> module %s does not exist or has a typo. Correct modules are: %s", selectedModule, getModuleNames())
```

Comment:
```suggestion
			return fmt.Errorf("module %s does not exist or has a typo. Correct modules are: %s", selectedModule, getModuleNames())
```
These error prefixes are not needed, just make sure the rest of the message is specific enough.